genesis_seed/dm/models.py

```diff
@@ -71,3 +172,156 @@ class Node(SimpleViewMixin):
     project_id: sys_uuid.UUID = c.SERVICE_PROJECT_ID
     name: str = ""
     description: str = ""
+
+    def __post_init__(self):
+        if isinstance(self.uuid, str):
+            self.uuid = sys_uuid.UUID(self.uuid)
+
+        if isinstance(self.project_id, str):
+            self.project_id = sys_uuid.UUID(self.project_id)
+
+
+@dataclasses.dataclass
+class Payload(SimpleViewMixin):
+    machine: Machine | None = None
+    node: Node | None = None
+    interfaces: tp.List[Interface] = dataclasses.field(default_factory=list)
+
+    # This field is used for CP payloads
+    payload_hash: str = ""
+    payload_updated_at: str = dataclasses.field(
+        default_factory=lambda: datetime.datetime(
+            1970, 1, 1, 0, 1, tzinfo=datetime.timezone.utc
+        ).isoformat()
+    )
+
+    def __eq__(self, other: "Payload") -> bool:
+        return self.__hash__() == other.__hash__()
+
+    def __hash__(self) -> int:
+        if self.payload_hash:
+            return hash(self.payload_hash)
+
+        return hash(self.calculate_payload_hash())
+
+    def dump_to_simple_view(self):
+        view = super().dump_to_simple_view()
+
+        if self.machine is not None:
+            view["machine"] = self.machine.dump_to_simple_view()
+
+        if self.node is not None:
+            view["node"] = self.node.dump_to_simple_view()
+
+        view["interfaces"] = [i.dump_to_simple_view() for i in self.interfaces]
+
+        return view
+
+    def calculate_payload_hash(self) -> str:
+        """Calculate payload hash using dedicated fields."""
+        m = hashlib.sha256()
+        data = {}
+
+        # Base payload object
+        if self.machine is not None:
+            data = {
+                "machine": {
+                    "image": self.machine.image,
+                    "node": str(self.node),
+                }
+            }
+
+        if self.node is not None:
+            data["node"] = {
+                "cores": self.node.cores,
+                "ram": self.node.ram,
+                "node_type": self.node.node_type,
+                "image": self.node.image,
+            }
+
+        if self.interfaces:
+            data["interfaces"] = [
+                {
+                    "mac": iface.mac,
+                    "ipv4": None,
+                    "mask": None,
+                    "ipv4_addresses": [str(ip) for ip in iface.ipv4_addresses],
+                    "masks": [str(mask) for mask in iface.masks],
+                }
+                for iface in self.interfaces
+            ]
+
+        m.update(json.dumps(data).encode("utf-8"))
```

Comment:
Minor, generate minimized json without spaces:

```suggestion
        m.update(json.dumps(data, separators=(',', ':')).encode("utf-8"))
```